jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -134,10 +135,14 @@ public HttpURI(String scheme, String host, int port, String path, String param,
         _scheme = scheme;
         _host = host;
         _port = port;
-        parse(State.PATH, path, 0, path.length());
-        _param = param;
-        _query = query;
-        _fragment = fragment;
+        if (path != null)
+            parse(State.PATH, path, 0, path.length());
+        if (param != null)
+            _param = param;
+        if (query != null)
+            _query = query;
+        if (fragment != null)
+            _fragment = fragment;
```

Comment:
I would not do null checks for just assignments.
Path needs null check because `path.length()` may NPE.

Reply:
I'm doing null checks because if we parse the path, we may discover params, fragments and queries in it.    So either we through if they are non null, or just not overwrite them unless we have explicit non null values.